src/test/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/model/sar/SARConditionFormatterTest.kt

```diff
@@ -0,0 +1,124 @@
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import uk.gov.justice.digital.hmpps.hmppshdcapi.model.sar.SARConditionFormatter.camelToSentenceCase
+import uk.gov.justice.digital.hmpps.hmppshdcapi.model.sar.SARConditionFormatter.getConditionText
+import uk.gov.justice.digital.hmpps.hmppshdcapi.model.sar.SARConditionFormatter.getConditionValues
+import uk.gov.justice.digital.hmpps.hmppshdcapi.model.sar.SARConditionFormatter.getPolicyVersion
+import uk.gov.justice.digital.hmpps.hmppshdcapi.model.sar.SARConditionFormatter.policyVersions
+
+class SARConditionFormatterTest {
+
+  @Test
+  fun `camelCase is converted to sentence case`() {
+    assertThat("licenceInCvl".camelToSentenceCase()).isEqualTo("Licence in cvl")
+  }
+
+  @Test
+  fun `snake_case is converted to sentence case`() {
+    assertThat("licence_in_cvl".camelToSentenceCase()).isEqualTo("Licence in cvl")
+  }
+
+  @Test
+  fun `acronyms are handled correctly`() {
+    assertThat("JSONDataVersion".camelToSentenceCase()).isEqualTo("Json data version")
+  }
+
+  @Test
+  fun `numbers are separated correctly`() {
+    assertThat("version2".camelToSentenceCase()).isEqualTo("Version 2")
+    assertThat("JSONDataVersion2".camelToSentenceCase()).isEqualTo("Json data version 2")
+  }
+
+  @Test
+  fun `already spaced string is normalized`() {
+    assertThat("Licence in cvl".camelToSentenceCase()).isEqualTo("Licence in cvl")
+  }
+
+  @Test
+  fun `empty string returns empty`() {
+    assertThat("".camelToSentenceCase()).isEqualTo("")
+  }
+
+  @Test
+  fun `single word is capitalized`() {
+    assertThat("licence".camelToSentenceCase()).isEqualTo("Licence")
+  }
+
+  @Test
+  fun `mixed camel and snake case is handled`() {
+    assertThat("licence_inCvlData".camelToSentenceCase()).isEqualTo("Licence in cvl data")
+  }
+
+  @Test
+  fun `returns null for empty map`() {
+    val result = policyVersions.getConditionValues(emptyMap())
+    assertThat(result).isNull()
+  }
+
+  @Test
+  fun `converts multiple mixed keys`() {
+    val input = mapOf(
+      "licenceInCvl" to true,
+      "standard_conditions" to false,
+      "JSONDataVersion2" to 2,
+    )
+    val result = policyVersions.getConditionValues(input)
+    assertThat(result).containsExactlyInAnyOrderEntriesOf(
+      mapOf(
+        "Licence in cvl" to true,
+        "Standard conditions" to false,
+        "Json data version 2" to 2,
+      ),
+    )
+  }
+
+  @Test
+  fun `getConditionText with version returns correct text`() {
+    val (version, conditions) = policyVersions.entries.first()
+    val result = policyVersions.getConditionText(version, conditions[0].id)
+    assertThat(result).isEqualTo(conditions[0].text)
+  }
+
+  @Test
+  fun `getConditionText with version returns fallback if not found`() {
+    val (version) = policyVersions.entries.first()
+    val result = policyVersions.getConditionText(version, "nonexistent-id")
+    assertThat(result).isEqualTo("nonexistent-id")
+  }
```

Comment:
Is it worth having a version of this test that passes in `null` as the version, with the same expected outcome?